tests/e2e/common.go

```diff
@@ -103,6 +103,12 @@ type Output struct {
 	Status     interface{}       `json:"status,omitempty"`
 }
 
+type PipelineRolloutInfo struct {
+	PipelineRolloutName          string `json:"pipelineRolloutName"`
+	PipelineIsFailed             bool   `json:"pipelineIsFailed,omitempty"`
+	OverrideSourceVertexReplicas bool   `json:"overrideSourceVertexReplicas,omitempty"`
+}
```

Comment:
I don't think we really need the json tags for anything (we probably wouldn't serialize it as json), but let's just plan on removing that later if there are no other issues holding up this PR

Reply:
in a quick follow up PR can you remove the json tags, and also add some TODO comments in functional_test.go?